olp-cpp-sdk-dataservice-read/tests/StreamLayerClientImplTest.cpp

```diff
@@ -139,17 +216,16 @@ TEST(StreamLayerClientImplTest, Subscribe) {
     ASSERT_EQ(future.wait_for(kTimeout), std::future_status::ready);
 
     const auto& response = future.get();
-    ASSERT_TRUE(response.IsSuccessful());
+    EXPECT_TRUE(response.IsSuccessful());
```

Comment:
This was for a reason so that you don't continue and possibly crash as there is no result to get.
Why did you change to expect?

Reply:
It would be great to use assert here, but the problem is we write one test per one API method and use Scoped Traces for actually different tests. And that is why we should use ASSERT only where we can not use EXPECT, because in other case next scoped parts of the test will not run.

In this particular and similar cases there is no problem with EXPECT, because ApiResponse has member of ResultType in any case, even if IsSuccessful()==false. If ApiResponse contains an error, GetResult() will return default constructed ResultType value. And further checks will be focused on this default constructed result value, and this is ok.

So we can use EXPECT here and so we should us it to increase chances other sub-tests to run.

Reply:
Usually, this should not be so black and white on unit tests as you should only be able to break this during development if you really work on the tested method. I would assume this to be true on integration test.